code/game/objects/random/materials.dm

```diff
@@ -2,6 +2,7 @@
 	name = "random building material"
 	icon_state = "material-grey"
 
+//This stuff can't be easily converted to pickweight because of these amount fields
```

Comment:
For consistency's sake, can't you just make a new obj that's a child of these materials that just has the amount changed on it?

Reply:
i'll do something about it, DNM for now